pkg/gather/gather_test.go

```diff
@@ -392,18 +394,18 @@ func assertMetadataOneGatherer(
 	err := json.Unmarshal(metadataBytes, &archiveMetadata)
 	assert.NoError(t, err)
 
-	for i := range archiveMetadata.StatusReports {
-		archiveMetadata.StatusReports[i].Duration = 0
+	for _, v := range archiveMetadata.StatusReports {
+		v.Duration = 0
```

Comment:
It's actually gonna fail if the test gatherer takes >= 1 ms because this expression assigns the value to a copy of the status report, you need to do something like `archiveMetadata.StatusReports[key].Duration = 0`. Go sometimes isn't obvious..